### DISCUSSION TITLE
Use NO (or other keywords) as slot name

### DISCUSSION THREAD

**Discussion by joepvgenuchten** [linkml/linkml#2160] [Q&A] [answered]
Hi, 

For reasons that are not in my control (legacy compatibility) I need to define schema's that have NO as an attribute/column name.  I'm not sure if NO is a yaml keyword or a linkML keyword, but when I try define a slot called NO,  linkML interprets it as 'False' and runs into all kinds of errors (i'm guessing because it doesnt expect a what it interprets as a boolean literal to be in the place where it expects a slot (name)). 

- Is there a way yo use the NO as a slot name directly? (tell the interpreter to overwrite the False interpretation)?
- If not: Is there a way (maybe using aliases?) to name the slot something else but have LinkML render all its schemas using NO as attribute/column name?

any insight is much appreciated.

**Comment by dalito** [comment, marked as answer]
Yes, NO is a yaml keyword. Did you try `"NO"`instead of `NO`?

**Reply by joepvgenuchten** [reply]
That worked. Can you force all (class, slot, enum) name interpretation like that?

**Reply by dalito** [reply]
LinkML uses a yaml parser. which is conforming to YAML 1.1 that has this problematic parsing of boolean defined as part of the standard. yaml 1.2 only parses true/false as boolean and would have at least prevented your issue. There is already a related issue: https://github.com/linkml/linkml/issues/1883
